File: chapter_8/challenge2.c
// Challenge 2, Chapter 8
// jeffreydiaz27@gmail.com

#include <readline/readline.h>
#include <stdio.h>

int main(int argc, const char* argv[])
{
	printf("Where should I start counting? ");
	const char* input = readline(NULL);
	int num = atoi(input);

	if (num) {
		for (int i = num; i >= 0; i -= 3) {
			printf("%d\n", i);
			if ((i % 5) == 0)
				printf("Found one!\n");
		}
	} else {
		printf("CYAA\n");
	}

	return 0;
}

File: chapter_16/challenge2.m
// Challenge 2, Chapter 16
// jeffreydiaz27@gmail.com

#import <Foundation/Foundation.h>
#import <readline/readline.h>

int main(int argc, const char* argv[])
{
	@autoreleasepool {
		NSLog(@"Where should I start counting? ");
		const char* input = readline(NULL);
		NSString* input_str = [NSString stringWithUTF8String:input];
		
		int num = atoi([input_str UTF8String]);
		
		if (num) {
			for (int i  = num; i >= 0; i -= 3) {
				NSLog(@"%d\n", i);
				if ((i % 5) == 0)
					NSLog(@"Found one!\n");
			}
		} else {
			NSLog(@"CYYA\n");
		}
	}
	return 0;
}
